### PR TITLE
feat: Display blocked user properly

### PR DESCRIPTION
**Pull request by borichellow**
# Second part of [Task-1593](https://wearezeta.atlassian.net/browse/AR-1593)
 there will be one more. Also notice that the target merge branch is not `develop` but `feat/block_user_from_conversation_list`, so it will be merged one by one and we'll get I big PR for the whole feature.

# What's new in this PR?

Blocked user is displayed properly in all the places (search, conversation, conversation participants, profile, etc.)

### Attachments (Optional)

photo_5424655604972110844_y

photo_5424655604972110842_y

photo_5424655604972110843_y

photo_5424655604972110841_y

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-820/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-820/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-820/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-820/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4054796)(4054796)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4054796)(4054796)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-820/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4055166)(4055166)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4055166)(4055166)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt [hunk 1]
```diff
@@ -218,182 +228,294 @@ private fun MessageComposer(
...
+                    isFileSharingEnabled,
+                    tempCachePath
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun AttachmentOptions(
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt:301]: Maybe this is a good moment to rename and extract this to a separate `MessageAttachmentOptions` file?

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt [hunk 2]
```diff
@@ -218,182 +228,294 @@ private fun MessageComposer(
...
+                exit = fadeOut()
+            ) {
+                SendButton(
+                    isEnabled = messageComposerState.sendButtonEnabled,
+                    onSendButtonClicked = onSendButtonClicked
+                )
+            }
+        }
+    }
+}
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt:379]: Same than above. Can we move to a different file? Maybe one grouping all the MessageComposeActions ?

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt [hunk 3]
```diff
@@ -218,182 +228,294 @@ private fun MessageComposer(
...
+            CollapseIconButton(
+                onCollapseClick = { messageComposerState.toggleFullScreen() },
+                collapseRotation = collapseButtonRotationDegree
+            )
         }
     }
 }
 
+@Composable
+private fun UserIsBlockedMessage() {
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt:485]: As this is not really a conditional compose object, my suggestion would be to rename it to `BlockedUserMessage` and move it to a separate file.

**Review by gongracr — CHANGES_REQUESTED** [review]
I added a few non-important comments, mostly about separating code concerns to other files. But massive work here 💪

File: app/src/main/kotlin/com/wire/android/ui/common/UserBadge.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.wire.android.ui.home.conversationslist.model.Membership
+import com.wire.android.ui.home.conversationslist.model.hasLabel
+import com.wire.kalium.logic.data.user.ConnectionState
+
+@Composable
+fun UserBadge(membership: Membership, connectionState: ConnectionState? = null, startPadding: Dp = 0.dp) {
```
[the remark this answers is not in the document]
Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/ui/common/UserBadge.kt:15]: good idea

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-820/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4099196)(4099196)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4099196)(4099196)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageActionsBox.kt [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+private fun MessageComposeActions(
+    messageComposerState: MessageComposerInnerState,
+    focusManager: FocusManager
+) {
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceEvenly,
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(56.dp)
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageActionsBox.kt:61]: Magic number?

**Review by gongracr — APPROVED** [review]
I added some tiny comments. But looking in great shape. Thx for the changes 🙏 🥮 ☕

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-820/8/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4100473)(4100473)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4100473)(4100473)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-820/10/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4115461)(4115461)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4115461)(4115461)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt [hunk 6]
```diff
@@ -226,9 +223,12 @@ private fun MessageContent(
...
         )
-        is MessageContent.SystemMessage.MemberAdded -> {}
-        is MessageContent.SystemMessage.MemberLeft -> {}
-        is MessageContent.SystemMessage.MemberRemoved -> {}
+        is MessageContent.SystemMessage.MemberAdded -> {
+        }
+        is MessageContent.SystemMessage.MemberLeft -> {
+        }
+        is MessageContent.SystemMessage.MemberRemoved -> {
+        }
```
[the remark this answers is not in the document]
Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt:231]: yes, finally i updated kotlin plugin and now it doesn't change empty `{}` :)

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-820/11/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4115623)(4115623)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4115623)(4115623)-dev-release.apk)

**Pull request merged**